File: 2598-shortest-distance-to-target-string-in-a-circular-array/shortest-distance-to-target-string-in-a-circular-array.java
class Solution {
    public int closestTarget(String[] words, String target, int startIndex) {
        int res = Integer.MAX_VALUE, n = words.length;
        for(int i=0; i<words.length; i++) {
            if(words[i].equals(target)) {
                int len1 = Math.abs(startIndex - i) ;
                int len2 = n - len1;
                res = Math.min(res, Math.min(len1, len2));
            }
        }
        return (res == Integer.MAX_VALUE) ? -1 : res;
    }
}

File: 2598-shortest-distance-to-target-string-in-a-circular-array/shortest-distance-to-target-string-in-a-circular-array.go
func closestTarget(words []string, target string, startIndex int) int {
    res := math.MaxInt;
    n := len(words);
    for i := 0; i < n; i++ {
        if words[i] == target {
            len1 := startIndex - i;
            if len1 < 0 {
                len1 = -len1;
            }
            len2 := (n - len1);
            res = min(res, min(len1, len2));
        }
    }
    if res == math.MaxInt {
        return -1
    }
    return res  
}